core/src/main/java/org/testcontainers/containers/wait/strategy/HttpWaitStrategy.java

```diff
@@ -108,6 +109,17 @@ public HttpWaitStrategy withBasicCredentials(String username, String password) {
         return this;
     }
 
+    /**
+     * Set the HTTP connections read timeout.
+     *
+     * @param timeout the timeout in millis
+     * @return this
+     */
+    public HttpWaitStrategy withReadTimeout(int timeout) {
```

Comment:
Could you please make it `Duration`?

Reply:
Done, the only problem is that you are possibly passing `Duration.ofNanos()`, but `setReadTimeout()` is not able to handle nanos, but millis.

Reply:
I would add a guard that checks that after calling `toMillis()` the amount is positive

Reply:
done